app/assets/javascripts/common/reports.js

```diff
@@ -458,8 +457,11 @@ Reports = function (withLtfu) {
             },
           },
         },
-      }
+      },
     };
+    if (isGoalLineEnabled(document.getElementById("bp-controlled"))) {
+      config = addGoalLineToConfig(config, controlledGraphRate);
+    }
 
```

Comment:
isGoalLineEnabled(document.getElementById("bp-controlled"))
Takes a boolean (its a string of the boolean) from data-show-goal-lines in the show.erb.html
This function takes the whole card and returns the 'data-..' as a real boolean

addGoalLineToConfig(config, controlledGraphRate)
This takes in the receives the current config and all the values for the chart to process the goal and return new config object which includes the additional goal line plugin settings